lib/datadog/core/error_tracking/component.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require_relative 'collector'
+require_relative 'filters'
+
+module Datadog
+  module Core
+    module ErrorTracking
+      # Component for error tracking.
+      #
+      # Only one instance of the Component should ever be active.
+      #
+      # The component instance records every handled exceptions from the configured scopes
+      # (user, third_party packages, specified modules of everything).
+      class Component
+        attr_accessor :handled_exc_tracker,
+          :module_path_getter,
+          :tracer,
+          :modules_to_instrument,
+          :instrumented_files,
+          def self.build(settings, tracer)
+            if settings.error_tracking.instrumentation_scope.empty? && settings.error_tracking.modules_to_instrument.empty?
```

Comment:
If this is the only check, it can probably be done outside of this class, in the caller, so that this class's initializer is trivia, and only invoked when the values are valid.

Reply:
addressed in ecfdce57e4880c87d13c07923e31ef91a92f64e7